messages/teleporter/message_handler.go

```diff
@@ -167,7 +183,47 @@ func (m *messageHandler) ShouldSendMessage(destinationClient vms.DestinationClie
 		return false, nil
 	}
 
-	return true, nil
+	var decision bool = true
+	deciderClientValue := reflect.ValueOf(m.deciderClient)
+	if deciderClientValue.IsValid() && !deciderClientValue.IsNil() {
+		warpMsgIDStr := m.unsignedMessage.ID().Hex()
+
+		warpMsgID, err := hex.DecodeString(warpMsgIDStr)
+		if err != nil {
+			m.logger.Error(
+				"Error decoding message ID",
+				zap.String("warpMsgIDStr", warpMsgIDStr),
+				zap.Error(err),
+			)
+			return decision, err
+		}
+
+		// TODO: add a timeout to the context
+		response, err := m.deciderClient.ShouldSendMessage(
+			context.Background(),
+			&pbDecider.ShouldSendMessageRequest{
+				NetworkId:           m.unsignedMessage.NetworkID,
+				SourceChainId:       m.unsignedMessage.SourceChainID[:],
+				Payload:             m.unsignedMessage.Payload,
+				BytesRepresentation: m.unsignedMessage.Bytes(),
+				Id:                  warpMsgID,
+			},
+		)
+		if err != nil {
+			m.logger.Error(
+				"Error response from decider.",
+				zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+				zap.String("teleporterMessageID", teleporterMessageID.String()),
+				zap.Any("warpMessageID", warpMsgIDStr),
+				zap.Error(err),
+			)
+			return decision, err
+		}
```

Comment:
The code shifted around since this comment and I think the comment no longer applies. (the "Outdated" label is appropriate for once haha.) Let me know if that's not the case and if there's still some improvement that could be made around this.